### PR TITLE
Adding streamtracer

### PR DESCRIPTION
**Pull request by jgieseler**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jgieseler** [comment]
I confirm that I'm willing to be listed in the maintainer section.

**Comment by jgieseler** [comment]
https://github.com/orgs/conda-forge/teams/help-python, ready for review!

File: recipes/streamtracer/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  imports:
+    - streamtracer
+  commands:
+    - pip check
+  requires:
+    - pip
+
+about:
+  home: https://streamtracer.readthedocs.io/en/latest/
+  summary: Python wrapped fortran to calculate streamlines
```
Comment (ocefpaf) [inline recipes/streamtracer/meta.yaml:43]: Fortran? Is this summary correct? This recipe is using rust, no?

Comment (jgieseler) [reply recipes/streamtracer/meta.yaml:43]: I'm not the maintainer of the original code (but in contact), so I did not change any description from the original one. But yes, I think it was written in Fortran and is in the process of being moved to Rust.

File: recipes/streamtracer/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    - python
+    - setuptools >=45
+    - setuptools-scm >=6.2
+    - numpy
+    - maturin >=0.13,<0.14
+    - pip
+  run:
+    - python
+    - numpy >=1.14.5
+    - libgcc-ng  # [linux]
```
[the remark this answers is not in the document]
Comment (jgieseler) [reply recipes/streamtracer/meta.yaml:31]: Not 100% sure, but I think I do. At least when I was testing locally on Ubuntu, I needed it. And the [Azure testing was complaining without it](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=756871&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1335).

[the remark this answers is not in the document]
Comment (ocefpaf) [reply recipes/streamtracer/meta.yaml:31]: @isuruf I thought that adding the c-compiler would "fix this." Do you have any advice for us here?

[the remark this answers is not in the document]
Comment (ocefpaf) [reply recipes/streamtracer/meta.yaml:31]: @jgieseler your previous commits also had more changes, like the noarch one, can you try to remove it again so we can check with this "version" of the recipe?

[the remark this answers is not in the document]
Comment (jgieseler) [reply recipes/streamtracer/meta.yaml:31]: I think you can still see the Azure logs of that version at https://github.com/conda-forge/staged-recipes/pull/23569/checks?check_run_id=15622002783, should I really go back to that version?

File: recipes/streamtracer/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  host:
+    - python
+    - setuptools >=45
+    - setuptools-scm >=6.2
+    - numpy
+    - maturin >=0.13,<0.14
+    - pip
+  run:
+    - python
+    - numpy >=1.14.5
```
Comment (ocefpaf) [inline recipes/streamtracer/meta.yaml:30]: B/c numpy is used during the build you need this here.

```suggestion
    - {{ pin_compatible('numpy') }}
```

**Comment by jgieseler** [comment]
Thanks a lot @ocefpaf!

**Pull request merged**